evm/spec/framework.tex

```diff
@@ -151,4 +110,46 @@ \subsubsection{Constraints}
+
+In other words, we need to ensure that the rows -- reduced to the input and output columns -- of $S_1$ calling $Op$ are permutations of the rows of $S_2$ that carry out $Op$. Our CTL protocol is also based on logUp and is similar to our range-checks.
+
+To prove this, the first step is to only select the rows of interest in $S_1$ and $S_2$, and filter out the rest. Let $f^1$ be the filter for $S_1$ and $f^2$ the filter for $S_2$. $f^1$ and $f^2$ are constrained to be in $\{0, 1\}$. $f^1 = 1$ (resp. $f^2 = 1$) whenever the row at hand carries out $Op$ in $S_1$ (resp. in $S_2$), and 0 otherwise. Let also $(\alpha, \beta)$ be two random challenges.
+
+The idea is to create subtables $S_1'$ and $S_2'$ of $S_1$ and $S_2$ respectively, such that $f^1 = 1$ and $f^2 = 1$ for all their rows. The columns in the subtables are limited to the ones whose values must be identical (the inputs and outputs of $Op$ in our example).
+
+Let $\{c^{1, i}\}_{i=1}^m$ be the columns in $S_1'$ an $\{c^{2,i}\}_{i=1}^m$ be the columns in $S_2'$.
+
+The prover defines a ``running sum'' $Z$ for $S_1'$ such that:
+\begin{gather*}
+  Z^{S_1}_{n-1} = \frac{1}{\sum_{j=0}^{m-1} \alpha^j \cdot c^{1, j}_{n-1} + \beta} \\
+  Z^{S_1}_{i+1} = Z^{S_1}_i + f^1_i \cdot \frac{1}{\sum_{j=0}^{m-1} \alpha^j \cdot c^{1, j}_i + \beta}
+\end{gather*}
+The second equation ``selects'' the terms of interest thanks to $f^1$ and filters out the rest.
+
+Similarly, the prover constructs a running sum $Z^{S_2}$for $S_2$. Note that $Z$ is computed ``upside down'': we start with $Z_{n-1}$ and the final sum is in $Z_0$. 
+
+On top of the constraints to check that the running sums were correctly constructed, the verifier checks that $Z^{S_1}_0 = Z^{S_2}_0$.
+This ensures that the columns in $S_1'$ and the columns in $S_2'$ are permutations of each other.
+
+In other words, the CTL argument is a logUp lookup argument where $S_1'$ is the looking table, $S_2'$ is the looked table, and $S_1' = S_2'$ (all the multiplicities are 1).
+
+To sum up, for each STARK $S$, the prover:
+\begin{enumerate}
+  \item constructs a running sum $Z_i^l$ for each table looking into $S$ (called looking sums here),
+  \item constructs a running sum $Z^S$ for $S$ (called looked sum here),
+  \item sends the final value for each running sum $Z_{i, 0}^l$ and $Z^S_0$ to the verifier,
+  \item sends a commitment to $Z_i^l$  and $Z^S$ to the verifier.
+\end{enumerate}
+Then, for each STARK $S$, the verifier:
+\begin{enumerate}
+  \item computes the sum $Z = \sum_i Z_{i, 0}^l$,
+  \item checks that $Z = Z^S_0$,
+  \item checks that each $Z_i^l$  and $Z^S$ was correctly constructed.
```

Comment:
can't it be left to its original position?

Reply:
It could, but since logUp is described in details in the Range-check section I wanted to move the CTLs afterwards.

Reply:
Seems to me like CTLs are much more core than RCs, and should hence be introduced where they were originally (eventually with a footnote saying logUp is described in more depth later), but up to you